registry/txt.go

```diff
@@ -191,8 +191,11 @@ func (im *TXTRegistry) Records(ctx context.Context) ([]*endpoint.Endpoint, error
 			if plan.IsManagedRecord(ep.RecordType, im.managedRecordTypes, im.excludeRecordTypes) {
 				// Get desired TXT records and detect the missing ones
 				desiredTXTs := im.generateTXTRecord(ep)
-				for _, desiredTXT := range desiredTXTs {
-					if _, exists := txtRecordsMap[desiredTXT.DNSName]; !exists {
+				// Exactly two records are old- and new- style, respectively
+				if len(desiredTXTs) == 2 {
```

Comment:
Tests are missing. Do you need help with them?

Reply:
@ivankatliarchuk FYI I haven't abandoned the PR: I'm planning to add them this week